src/DotVVM.Framework/Resources/Scripts/DotVVM.ts

```diff
@@ -1321,6 +1321,22 @@ class DotVVM {
             }
         };
 
+        ko.bindingHandlers["dotvvm-checkedItems"] = {
+            after: ko.bindingHandlers.checked.after,
+            init: ko.bindingHandlers.checked.init,
+            options: ko.bindingHandlers.checked.options,
+            update: function (element, valueAccessor, allBindings, viewModel, bindingContext) {
+                var value = valueAccessor();
+                if (!Array.isArray(valueAccessor()) && (!ko.isObservable(value) || !Array.isArray(value()))) {
+                    throw Error("The value of a `checkedItems` binding must be an array (i.e. not null nor undefined).");
+                }
+                if (ko.bindingHandlers.checked.update) {
+                    // although `checked` doesn't have an update, call it conditionally in case we ever add one
+                    return ko.bindingHandlers.checked.update(element, valueAccessor, allBindings, viewModel, bindingContext);
```

Comment:
I don't know how knockout handles `this` and if anything may go wrong, but you are calling `ko.bindingHandlers.checked.update` function in a different context than `ko.bindingHandlers.checked.options`. I'd be careful with that.